includes/class-wc-payments-webhook-processing-service.php

```diff
@@ -621,45 +629,41 @@ private function get_order_from_event_body_intent_id( $event_body ) {
 	}
 
 	/**
-	 * Gets the proper failure message from the code in the event.
+	 * Gets the proper failure message from the code in the error.
+	 * Error codes from https://stripe.com/docs/error-codes.
 	 *
-	 * @param array $event_body The event that triggered the webhook.
+	 * @param array $error The last payment error from the payment failed event.
 	 *
 	 * @return string The failure message.
 	 */
-	private function get_failure_message_from_event( $event_body ):string {
-		// Get the failure code from the event body.
-		$event_data    = $this->read_webhook_property( $event_body, 'data' );
-		$event_object  = $this->read_webhook_property( $event_data, 'object' );
-		$event_charges = $this->read_webhook_property( $event_object, 'charges' );
-		$charges_data  = $this->read_webhook_property( $event_charges, 'data' );
-		$failure_code  = $charges_data[0]['failure_code'] ?? '';
+	private function get_failure_message_from_error( $error ):string {
+		$code         = $error['code'] ?? '';
+		$decline_code = $error['decline_code'] ?? '';
+		$message      = $error['message'] ?? '';
 
-		switch ( $failure_code ) {
+		switch ( $code ) {
 			case 'account_closed':
-				$failure_message = __( "The customer's bank account has been closed.", 'woocommerce-payments' );
-				break;
+				return __( "The customer's bank account has been closed.", 'woocommerce-payments' );
 			case 'debit_not_authorized':
-				$failure_message = __( 'The customer has notified their bank that this payment was unauthorized.', 'woocommerce-payments' );
-				break;
+				return __( 'The customer has notified their bank that this payment was unauthorized.', 'woocommerce-payments' );
 			case 'insufficient_funds':
-				$failure_message = __( "The customer's account has insufficient funds to cover this payment.", 'woocommerce-payments' );
-				break;
+				return __( "The customer's account has insufficient funds to cover this payment.", 'woocommerce-payments' );
 			case 'no_account':
-				$failure_message = __( "The customer's bank account could not be located.", 'woocommerce-payments' );
-				break;
+				return __( "The customer's bank account could not be located.", 'woocommerce-payments' );
 			case 'payment_method_microdeposit_failed':
-				$failure_message = __( 'Microdeposit transfers failed. Please check the account, institution and transit numbers.', 'woocommerce-payments' );
-				break;
+				return __( 'Microdeposit transfers failed. Please check the account, institution and transit numbers.', 'woocommerce-payments' );
 			case 'payment_method_microdeposit_verification_attempts_exceeded':
-				$failure_message = __( 'You have exceeded the number of allowed verification attempts.', 'woocommerce-payments' );
-				break;
-
-			default:
```

Comment:
I wonder if we need here the `default:` case with `The payment was not able to be processed` failure message? I think it might be helpful if Stripe adds another code that we are not aware of

Reply:
It's actually covered with [this new return](https://github.com/Automattic/woocommerce-payments/blob/d26264a205f503b5691c07e263302117c7f35380/includes/class-wc-payments-webhook-processing-service.php#L667). And instead of returning just `The payment was not able to be processed` I also added Stripe's error message.

Maybe it's a bit confusing because I rewrote the switch to use `return` rather than `break`. So using a single return at the end will act as the `default`. I prefer it when we only want to return something, and especially if we have a nested `switch`.

Reply:
Aaaah, agree, makes sense.